app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -410,9 +410,11 @@ export default Ember.Component.extend({
       },
 
       onKeyUp(text, cp) {
-        const matches = /(?:^|[^a-z])(:(?!:).?[\w-]*:?(?!:)(?:t\d?)?:?) ?$/gi.exec(
-          text.substring(0, cp)
-        );
+        const regex = self.siteSettings.enable_inline_emoji_translation
```

Comment:
Given we are essentially duplicating the regex her with a minor difference, could you try to have two variables with explicit names, and combine them when needed please?